app/src/fs-utils.es6

```diff
@@ -1,9 +1,50 @@
 import fs from 'fs'
-import Utils from './flux/models/utils'
+import path from 'path';
 
-export function atomicWriteFileSync(filepath, content) {
-  const randomId = Utils.generateTempId()
-  const backupPath = `${filepath}.${randomId}.bak`
-  fs.writeFileSync(backupPath, content)
-  fs.renameSync(backupPath, filepath)
+function getSortedTimestampedFilesSync(filepath, basename, extension) {
+  let files = fs.readdirSync(filepath);
+  files = files.filter((file) => {
+    const hasConfig = file.indexOf(`${basename}.`) > -1;
+    const hasJSON = file.indexOf(`.${extension}`) > -1;
+    const hasExactName = file.indexOf(`${basename}.${extension}`) > -1;
+    return hasConfig && hasJSON && !hasExactName;
+  });
+
+  files = files.sort((a, b) => {
+    const aValues = a.split('.');
+    const bValues = b.split('.');
+
+    return parseInt(aValues[1], 10) - parseInt(bValues[1], 10);
+  });
+
+  return files;
+}
+
+export function atomicWriteFileSync(filepath, basename, extension, content) {
+  let fileNum = 0;
+  let newFile;
+  while (typeof newFile === "undefined" || fs.existsSync(newFile)) {
+    const milliseconds = (new Date()).getTime() * 1000 + fileNum;
+    newFile = path.join(filepath, `${basename}.${milliseconds}.${extension}`);
+    fileNum++;
+  }
+  fs.writeFileSync(newFile, content);
+
+  const files = getSortedTimestampedFilesSync(filepath, basename, extension);
+
+  while (files.length > 10) {
```

Comment:
this hardcoded `10` is the number of config.timestamp.json files to keep. we may want to put this in a variable instead of inline hardcode. we may also might want this to be configurable? /shrug

I figured 10 would be enough when I wrote this; I haven't ran into a case when all 10 are corrupt